test/itkPhaseAnalysisSoftThresholdImageFilterTest.cxx

```diff
@@ -98,21 +105,59 @@ itkPhaseAnalysisSoftThresholdImageFilterTest( int argc, char* argv[] )
   EXERCISE_BASIC_OBJECT_METHODS( phaseAnalyzer, PhaseAnalysisSoftThresholdImageFilter,
     PhaseAnalysisImageFilter );
 
-  bool applySoftThreshold = true;
+  bool applySoftThreshold = static_cast< bool >( atoi( argv[3] ) );
   TEST_SET_GET_BOOLEAN( phaseAnalyzer, ApplySoftThreshold, applySoftThreshold );
 
-  /*PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType numOfSigmas;
+  PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType numOfSigmas =
+    static_cast< PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType >( atof( argv[4] ) );
   phaseAnalyzer->SetNumOfSigmas( numOfSigmas );
-  TEST_SET_GET_VALUE( numOfSigmas, phaseAnalyzer->GetNumOfSigmas() );*/
-
-  /*std::cout << phaseAnalyzer->GetMeanAmp() << std::endl;
-  std::cout << phaseAnalyzer->GetSigmaAmp() << std::endl;
-  std::cout << phaseAnalyzer->GetThreshold() << std::endl;*/
+  TEST_SET_GET_VALUE( numOfSigmas, phaseAnalyzer->GetNumOfSigmas() );
 
   phaseAnalyzer->SetInput( vecInverseFFT->GetOutput() );
 
   TRY_EXPECT_NO_EXCEPTION( phaseAnalyzer->Update() );
 
+
+  // Regression tests
+  PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType expectedMeanAmp =
+    static_cast< PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType >( atof( argv[5] ) );
+  PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType computedMeanAmp =
+    phaseAnalyzer->GetMeanAmp();
+  if( itk::Math::NotAlmostEquals( expectedMeanAmp, computedMeanAmp ) )
+    {
+    std::cerr << "Test failed!" << std::endl;
+    std::cerr << "Error in GetMeanAmp()" << std::endl;
+    std::cerr << "Expected: " << expectedMeanAmp << ", but got: "
+      << computedMeanAmp << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType expectedSigmaAmp =
+    static_cast< PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType >( atof( argv[6] ) );
+  PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType computedSigmaAmp =
+    phaseAnalyzer->GetSigmaAmp();
+  if( itk::Math::NotAlmostEquals( expectedSigmaAmp, computedSigmaAmp ) )
+    {
+    std::cerr << "Test failed!" << std::endl;
+    std::cerr << "Error in GetSigmaAmp()" << std::endl;
+    std::cerr << "Expected: " << expectedSigmaAmp << ", but got: "
+      << computedSigmaAmp << std::endl;
+    return EXIT_FAILURE;
+    }
+
+  PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType expectedThreshold =
+    static_cast< PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType >( atof( argv[7] ) );
+  PhaseAnalysisSoftThresholdFilterType::OutputImagePixelType computedThreshold =
+    phaseAnalyzer->GetThreshold();
+  if( itk::Math::NotAlmostEquals( expectedThreshold, computedThreshold ) )
+    {
+    std::cerr << "Test failed!" << std::endl;
+    std::cerr << "Error in GetThreshold()" << std::endl;
+    std::cerr << "Expected: " << expectedThreshold << ", but got: "
+      << computedThreshold << std::endl;
+    return EXIT_FAILURE;
```

Comment:
Can you instead of exiting the test immediately, use a variable: `bool testPassed = true;` And then return EXIT_FAILURE or SUCCESS based on that variable? The good thing about this is that all parts of the test are executed, so we can correct it all at once, or have a more general idea what's going on.

See for example: https://github.com/phcerdan/ITKIsotropicWavelets/blob/0506972375509bfaf5b228ee108477044c6fcd2d/test/itkHeldIsotropicWaveletTest.cxx#L80